hawkauthlib/__init__.py

```diff
@@ -110,41 +127,20 @@ def get_signature(request, key, algorithm=None, params=None, accept_untrusted_co
         algorithm = "sha256"
     if params is None:
         params = utils.parse_authz_header(request, {})
-
-    if accept_untrusted_content is True:
-        # supports client-side flow, and old insecure functionality identified in Issue #6
-        sigstr = utils.get_normalized_request_string(request, params)
-    else:
-        # If the client has provided an optional hash Hawk parameter we will calculate on the server
-        # a coresponding hash to verify the signature properly
-        server_hash = "" if params.get("hash") is None else hash_payload(request, params, algorithm)
-        sigstr = utils.get_normalized_request_string(request, params, server_hash)
-    # The spec mandates that ids and keys must be ascii.
-    # It's therefore safe to encode like this before doing the signature.
-    sigstr = sigstr.encode("ascii")
-    if not isinstance(key, utils.bytes):
-        key = key.encode("ascii")
-    hashmod = ALGORITHMS[algorithm]
-    return utils.b64encode(hmac.new(key, sigstr, hashmod).digest())
+    if include_payload_hash:
+        params = params.copy()
+        params["hash"] = hash_payload(request, algorithm)    
+    return utils.get_signature(request, key, params, algorithm)
 
 
 @utils.normalize_request_object
-def hash_payload(request, params=None, algorithm=None):
-    """Generate the request payload hash on the server.
```

Comment:
There's nothing server-side-specific about this, so I removed that from the docs. I also removed the unused `params` parameter.